clientv3/ordering/util.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ordering
+
+import (
+	"sync"
+
+	"github.com/coreos/etcd/clientv3"
+)
+
+type KvOrderViolationFunc func(op clientv3.Op, resp clientv3.OpResponse, prevRev int64)
+
+type TxnOrderViolationFunc func(resp *clientv3.TxnResponse, prevRev int64)
```

Comment:
Build a `OpTxn` on `Commit`, then issue the RPC with `Do` to get an `OpResponse`?